experimental/framework/tinylicious-client/src/test/TinyliciousClient.spec.ts

```diff
@@ -5,42 +5,364 @@
+        const containerAndServices = ResettableTinyliciousClient.createContainer(containerConfig, schema);
+        const errorFn = (error) => {
+            assert.notStrictEqual(error.message, undefined, "TinyliciousClient error is undefined");
+            return true;
+        };
+
+        await assert.rejects(
+            containerAndServices,
+            errorFn,
+            "TinyliciousClient can create a container without initialization",
+        );
+    });
+});
+
+describe("TinyliciousClient post-initialization", () => {
     before(() => {
-        const clientConfig: TinyliciousConnectionConfig = { port: 7070 };
-        TinyliciousClient.init(clientConfig);
+        ResettableTinyliciousClient.init();
     });
 
     let documentId: string;
+    const clientConfig: TinyliciousConnectionConfig = { port: 7070 };
     beforeEach(() => {
         documentId = uuid();
     });
 
-    it("can create a container successfully", async () => {
+    /**
+    * Scenario: test if TinyliciousClient can get a non-exiting container.
+    *
+    * Expected behavior: an error should be thrown when trying to get a non-exisitent container.
+    */
+    it("cannot load improperly created container (cannot load a non-existent container)", async () => {
         const containerConfig: TinyliciousContainerConfig = { id: documentId };
         const schema: ContainerSchema = {
             name: documentId,
             initialObjects: {
                 map1: SharedMap,
             },
         };
+        const containerAndServices = ResettableTinyliciousClient.getContainer(containerConfig, schema);
+        const errorFn = (error) => {
+            assert.notStrictEqual(error.message, undefined, "TinyliciousClient error is undefined");
+            return true;
+        };
 
-        const [container] = await TinyliciousClient.createContainer(containerConfig, schema);
+        await assert.rejects(
+            containerAndServices,
+            errorFn,
+            "TinyliciousClient can load a non-existent container",
+        );
+    });
 
+    /**
+    * Scenario: test if TinyliciousClient can be initialized with a port number specified.
+    *
+    * Expected behavior: an error should not be thrown nor should a rejected promise
+    * be returned.
+    */
+    it("can create a container successfully with port number specification", async () => {
+        ResettableTinyliciousClient.resetInstance();
+        ResettableTinyliciousClient.init(clientConfig);
+        const containerConfig: TinyliciousContainerConfig = { id: documentId };
+        const schema: ContainerSchema = {
+            name: documentId,
+            initialObjects: {
+                map1: SharedMap,
+            },
+        };
+
+        const containerAndServices = ResettableTinyliciousClient.createContainer(containerConfig, schema);
+        const errorFn = (error) => {
+            assert.notStrictEqual(error.message, undefined, "Container cannot be created");
```

Comment:
looks like this check doesn't make sense when we expect it not to reject. docs say the error is always skipped anyway, so it's probably better just to omit this (if we can't specify undefined, we can use an empty fn instead)

Reply:
Right, I agree that having the error handler can be misleading when we expect it not to throw, so will replace it with an empty fn. On the other hand, I was a bit confused by the docs with the error handler being skipped, so I tested it out and it seems like the error handler is actually being run.